functions/assignOfflineContact.ts

```diff
@@ -0,0 +1,141 @@
+import '@twilio-labs/serverless-runtime-types';
+import {
+  Context,
+  ServerlessCallback,
+  ServerlessFunctionSignature,
+} from '@twilio-labs/serverless-runtime-types/types';
+import {
+  responseWithCors,
+  bindResolve,
+  error400,
+  error500,
+  success,
+  send,
+} from '@tech-matters/serverless-helpers';
+// eslint-disable-next-line prettier/prettier
+import type { PromiseValue } from 'type-fest';
+
+const TokenValidator = require('twilio-flex-token-validator').functionValidator;
+
+type EnvVars = {
+  TWILIO_WORKSPACE_SID: string;
+  TWILIO_CHAT_TRANSFER_WORKFLOW_SID: string;
+};
+
+export type Body = {
+  targetSid?: string;
+  finalTaskAttributes: string;
+};
+
+// eslint-disable-next-line prettier/prettier
+type TaskInstance = PromiseValue<ReturnType<ReturnType<ReturnType<ReturnType<Context['getTwilioClient']>['taskrouter']['workspaces']>['tasks']>['fetch']>>;
+// eslint-disable-next-line prettier/prettier
+type WorkerInstance = PromiseValue<ReturnType<ReturnType<ReturnType<ReturnType<Context['getTwilioClient']>['taskrouter']['workspaces']>['workers']>['fetch']>>;
+
+const assignToAvailableWorker = async (
+  event: Body,
+  newTask: TaskInstance,
+) => {
+  const reservations = await newTask.reservations().list();
+  const reservation = reservations.find(r => r.workerSid === event.targetSid);
+
+  if (reservation) {
+    await reservation.update({ reservationStatus: 'accepted' });
```

Comment:
Is it possible that the worker could see this come up in the All Tasks list and accept it before this code runs?  What would happen?

Reply:
This takes a few miliseconds, I don't think that's possible, but we maybe we can "disable the UI" for that case.